contracts/src/LightClient.sol

```diff
@@ -189,6 +216,14 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
         LightClientState memory newState,
         IPlonkVerifier.PlonkProof memory proof
     ) external {
+        //revert if we're in permissionedProver mode and the permissioned prover has not been set
+        if (permissionedProverMode == true && msg.sender != permissionedProver) {
```

Comment:
Since we no longer allow permissioned prover mode to be true if there is no permissioned prover set we don't need to check if permissioned prover mode is enabled and checking `msg.sender` should be enough, shouldn't it?

Reply:
very true, good catch and should hopefully save on a bit of gas fees